### PR TITLE
Update documentation

### LINKED ISSUE #70: Update NEWS and DESCRIPTION
These need to be done before the CRAN update.

### PR DESCRIPTION
**Pull request by GreatEmerald**
Also includes deprecating create16dayts and removing raster as a dependency. (Might still want to keep it in suggests, though?)

Fixes #70, #38.

### CODE REVIEW

File: R/bfast-deprecated.R [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#' Deprecated functions in package \pkg{bfast}.
+#' 
+#' The functions listed below are deprecated and will be removed in the future.
+#' Please use the alternative functions as listed below.
+#' Help pages for deprecated functions are available via \code{help("<function>-deprecated")}.
+#' @name bfast-deprecated
+#' @keywords internal
+NULL
```
Comment (mirt001) [inline R/bfast-deprecated.R:8]: More of a curiosity, I'm sure this was intentional, but why is ```NULL``` necessary here?

File: DESCRIPTION [hunk 2]
```diff
@@ -1,45 +1,50 @@
...
 Depends:
     R (>= 3.0.0)
 Imports:
     graphics,
     stats,
-    strucchange,
+    strucchangeRcpp,
     zoo,
     forecast,
     sp,
```
Comment (mirt001) [inline DESCRIPTION:38]: do we absolutely need sp?

Comment (GreatEmerald) [reply DESCRIPTION]: We absolutely do not need sp.

File: R/bfastlite.R [hunk 3]
```diff
@@ -24,18 +24,26 @@
...
                    stl = c("none", "trend", "seasonal", "both"),
                    decomp=c("stlplus", "stl"), sbins=1, ...)
 {
   data_pp <- bfastpp(data, order = order,
                    lag = lag, slag = slag, na.action = na.action,
                    stl = stl, decomp=decomp, sbins=sbins)
-  breakpoints <- strucchange::breakpoints(formula=formula, data = data_pp, ...)
-  return(breakpoints)
+  breakpoints <- strucchangeRcpp::breakpoints(formula=formula, data = data_pp, ...)
+  Result <- structure(list(breakpoints, data_pp), class="bfastlite")
```
Comment (mirt001) [inline R/bfastlite.R]: I don't like that Result starts with a capital R. So I checked how it is in all the other "results" and it's a mess. I'll start a new issue about this issue.

File: R/bfast-deprecated.R (hunk 1 above)
Comment (GreatEmerald) [reply R/bfast-deprecated.R:8]: This is for Roxygen2, comments can't just hang in the air, they must always precede something.

File: R/bfast.R [hunk 4]
```diff
@@ -144,22 +144,35 @@
 #' @export bfast
 bfast <- function (Yt, h = 0.15, season = c("dummy", "harmonic", "none"), 
                    max.iter = 10, breaks = NULL, hpc = "none", level = 0.05,
-                   reg = c("lm", "rlm"), decomp=c("stlplus", "stl"),
+                   reg = c("lm", "rlm"), decomp=c("stl", "stlplus"),
```
Comment (mirt001) [inline R/bfast.R:147]:
```suggestion
                   reg = c("lm", "rlm"), decomp = c("stl", "stlplus"),
```

File: R/bfast.R [hunk 5]
```diff
@@ -144,22 +144,35 @@
...
   reg = match.arg(reg)
   if(!(reg %in% c("lm","rlm"))) stop("Regression method unknown, use either 'lm' or 'rlm'.")
   if(reg == "rlm") require(MASS)
   decomp = match.arg(decomp)
-  if(decomp == "stlplus" && !require("stlplus",quietly = T)) stop("Please install the stlplus package!")
+  if (decomp == "stl" && any(is.na(Yt))) {
+    warning("The stl() function cannot deal with missing values in the time series, falling back to decomp='stlplus'.")
+    decomp <- "stlplus"
+  }
+  if(decomp == "stlplus" && !require("stlplus",quietly = T)) {
```
Comment (mirt001) [inline R/bfast.R:159]:
```suggestion
  if(decomp == "stlplus" && !require("stlplus", quietly = T)) {
```

File: R/bfast.R [hunk 6]
```diff
@@ -144,22 +144,35 @@
...
   level  <- rep(level, length.out = 2)
   ti <- time(Yt)
   f <- frequency(Yt) # on cycle every f time points (seasonal cycle)
-  if (class(Yt) != "ts") 
+  if (class(Yt) != "ts")
     stop("Not a time series object")
+  if (f < 1)
+    stop("Time series frequency needs to be more than 1")
+  if (decomp == "stlplus" && f < 4) {
+    warning("Not enough seasons (frequency < 4) for using stlplus decomposition, falling back to decomp='stl'")
```
Comment (mirt001) [inline R/bfast.R:173]:
```suggestion
    warning("Not enough seasons (frequency < 4) for using stlplus decomposition, falling back to decomp = 'stl'")
```

File: NEWS [hunk 7]
```diff
@@ -1,3 +1,29 @@
+Changes in Version 1.6.0
+
+  o Integrated work by Marius Appel on speeding up BFAST algorithms using Rcpp.
+  Whether the speedups are enabled is controlled by set_fast_options() vs
+  set_default_options(). The fast options are now default.
```
Comment (appelmar) [inline NEWS:5]: We need to change this description once #9 is fixed (set_fallback_options).

File: examples/bfastlite.r [hunk 8]
```diff
@@ -5,9 +5,9 @@ tsp(datats) <- tsp(simts$time.series) # assign correct time series attributes
...
 # Detect breaks
-bp = bfast0n(datats)
+bp = bfastlite(datats)
 
 # Default method of estimating breakpoints
-bp[["breakpoints"]]
+bp[["breakpoints"]][["breakpoints"]]
 # Custom method
-strucchange::breakpoints(bp, breaks=2)
+strucchangeRcpp::breakpoints(bp[["breakpoints"]], breaks=2)
```
Comment (mirt001) [inline examples/bfastlite.r:13]:
```suggestion
strucchangeRcpp::breakpoints(bp[["breakpoints"]], breaks = 2)
```

File: examples/bfastts.r [hunk 9]
```diff
@@ -0,0 +1,6 @@
+# 16-day time series (i.e. MODIS)
+timedf <- data.frame(y=som$NDVI.b, dates=dates[1:nrow(som)])
+bfastts(timedf$y, timedf$dates, type="16-day")
+
+# Irregular
+head(bfastts(timedf$y, timedf$dates, type="irregular"), 50)
```
Comment (mirt001) [inline examples/bfastts.r:6]:
```suggestion
timedf <- data.frame(y = som$NDVI.b, dates = dates[1:nrow(som)])
bfastts(timedf$y, timedf$dates, type = "16-day")

# Irregular
head(bfastts(timedf$y, timedf$dates, type = "irregular"), 50)
```

**Review by mirt001 — CHANGES_REQUESTED** [review]
pretty much only spaces.

File: NEWS (hunk 7 above)
Comment (GreatEmerald) [reply NEWS:5]: Also the readme!

**Pull request merged**